js/runner.go

```diff
@@ -122,7 +122,45 @@ func (r *Runner) NewVU(samplesOut chan<- stats.SampleContainer) (lib.VU, error)
 	return lib.VU(vu), nil
 }
 
-func (r *Runner) newVU(samplesOut chan<- stats.SampleContainer) (*VU, error) {
+func getRandomIP(ifacesList string) (*net.IPAddr, error) {
+	var addresses []net.Addr
+
+	for _, ifaceName := range strings.Split(ifacesList, ",") {
+		targetIf, err := net.InterfaceByName(ifaceName)
+		if err != nil {
+			return nil, errors.New("Cannot find network interface '" + ifaceName + "'")
+		}
+
+		addrs, err := targetIf.Addrs()
+		if err != nil {
+			return nil, errors.New("Cannot get addresses on network interface '" + ifaceName + "'")
+		}
+
+		addresses = append(addresses, addrs...)
+	}
+
+	if len(addresses) == 0 {
+		return nil, errors.New("Cannot find any usable addresses on network interface(s) '" + ifacesList + "'")
+	}
+
+	rand.Shuffle(len(addresses), func(i, j int) { addresses[i], addresses[j] = addresses[j], addresses[i] })
+
+	tip, _, err := net.ParseCIDR(addresses[0].String())
+	if err != nil {
+		return nil, errors.New("Cannot parse IP address from string '" + addresses[0].String() + "'")
+	}
+
+	localAddr, err := net.ResolveIPAddr("ip", tip.String())
+	if err != nil {
+		return nil, errors.New("Cannot resolve IP '" + addresses[0].String() + "'")
+	}
+
+	return localAddr, nil
+}
+
+func (r *Runner) newVU(samplesOut chan<-: stats.SampleContainer) (*VU, error) {
```

Comment:
The syntax error is because of the colon.

```suggestion
func (r *Runner) newVU(samplesOut chan<- stats.SampleContainer) (*VU, error) {
```

Reply:
Syntax error fixed in 90bbf39bf6443feea1792eb169629e9eb61c129f. Thank you!